desdeo_tools/scalarization/Scalarizer.py

```diff
@@ -0,0 +1,123 @@
+"""Implements methods for scalarizing vector valued function.
+
+"""
+from typing import Callable, Optional, Any
+
+import numpy as np
+
+
+class Scalarizer:
+    """Implements a class for scalarizing vector valued functions with a
+    given scalarization function.
+    """
+
+    # How to scalarize?
+    # What can be expected from the function?
+
+    def __init__(
+        self,
+        evaluator: Callable,
+        scalarizer: Callable,
+        evaluator_args=None,
+        scalarizer_args=None,
+    ):
+        """
+        Args:
+            evaluator (Callable): A Callable object returning numpy array.
+            scalarizer (Callable): A function which should accepts as its
+            arguments the output of evaluator and return a single value.
+            evaluator_args (Any, optional): Optional arguments to be passed to
+            evaluator. Defaults to None.
+            scalarizer_args (Any, optional): Optional arguments to be passed to
+            scalarizer. Defaults to None.
+        """
+        self._evaluator = evaluator
+        self._scalarizer = scalarizer
+        self._evaluator_args = evaluator_args
+        self._scalarizer_args = scalarizer_args
+
+    def evaluate(self, xs: np.ndarray) -> np.ndarray:
+        """Evaluates the scalarized function with the given arguments and
+        returns a scalar value for each vector on variables given in a numpy
+        array.
+        
+        Args:
+            xs (np.ndarray): A 2D numpy array containing vectors of variables
+            on each of its rows.
+        
+        Returns:
+            np.ndarray: A 1D numpy array with the values returne by the
+            scalarizer for each row in xs.
+        """
+        if self._evaluator_args is not None:
+            res_eval = self._evaluator(xs, **self._evaluator_args)
+        else:
+            res_eval = self._evaluator(xs)
+
+        if self._scalarizer_args is not None:
+            res_scal = self._scalarizer(res_eval, **self._scalarizer_args)
+        else:
+            res_scal = self._scalarizer(res_eval)
+
+        return res_scal
+
+    def __call__(self, xs: np.ndarray) -> np.ndarray:
+        """Wrapper to the evaluate method.
+        """
+        return self.evaluate(xs)
+
+
+if __name__ == "__main__":
+    from desdeo_problem.Problem import MOProblem
+    from desdeo_problem.Objective import _ScalarObjective
+    from desdeo_problem.Variable import variable_builder
```

Comment:
I see why you are doing it, but I do not think that desdeo_tools should have desdeo_problem as a dependency. I have a feeling that, in the future, there will be a few things in this package that will be used in desdeo_problem. That will lead to circular dependency.

The classes that you wrote are general enough to not be limited to desdeo_problem. Can you shift these particular examples to desdeo_mcdm, replacing them with a very general/basic example here?

Reply:
My intention was to get rid of them since we decided not to have desdeo_problem as a dependency. I just forgot.